osu.Game.Rulesets.Osu/Difficulty/OsuLegacyScoreProcessor.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Rulesets.Osu.Difficulty
+{
+    internal class OsuLegacyScoreProcessor : ILegacyScoreProcessor
+    {
+        public int AccuracyScore { get; private set; }
+
+        public int ComboScore { get; private set; }
+
+        public double BonusScoreRatio => legacyBonusScore == 0 ? 0 : (double)modernBonusScore / legacyBonusScore;
+
+        private int legacyBonusScore;
+        private int modernBonusScore;
+        private int combo;
+
+        private double scoreMultiplier;
+        private IBeatmap playableBeatmap = null!;
+
+        public void Simulate(IWorkingBeatmap workingBeatmap, IBeatmap playableBeatmap, IReadOnlyList<Mod> mods)
+        {
+            this.playableBeatmap = playableBeatmap;
+
+            IBeatmap baseBeatmap = workingBeatmap.Beatmap;
+
+            int countNormal = 0;
+            int countSlider = 0;
+            int countSpinner = 0;
+
+            foreach (HitObject obj in workingBeatmap.Beatmap.HitObjects)
+            {
+                switch (obj)
+                {
+                    case IHasPath:
+                        countSlider++;
+                        break;
+
+                    case IHasDuration:
+                        countSpinner++;
+                        break;
```

Comment:
I can't use concrete types here because they're `ConvertHitObject`s which are all internal (e.g. https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Objects/Legacy/Osu/ConvertHit.cs). This being the un-converted beatmap.

For the most part this matches https://github.com/ppy/osu/blob/90cb3dac8601970ffe5c7c6a9f73a9c37bb185bd/osu.Game.Rulesets.Osu/Beatmaps/OsuBeatmapConverter.cs#L27-L76

Reply:
I see. Was mostly a passing remark anyways, not anything actionable.